api/api.go

```diff
@@ -178,38 +178,38 @@ func (s *authedRequester) Do(r *Request) *Response {
 		theurl.RawQuery = encodedParams
 	}
 	req, err := http.NewRequest(reqMethod, theurl.String(), bytes.NewReader(r.Body.Bytes()))
+	if err != nil {
+		return &Response{err: fmt.Errorf("failed to build HTTP request: %w", err)}
+	}
 
 	// Specify content type for POST requests, as the body format has to be specified
 	if r.ReqType == PostReq {
 		req.Header.Set("Content-Type", "application/x-www-form-urlencoded; param=value")
 	}
 
+	res, err := s.client.Do(req)
 	if err != nil {
-		return &Response{err: err}
-	}
-	client := &http.Client{}
-	res, err := client.Do(req)
-	if err != nil {
-		return &Response{err: err}
+		return &Response{err: fmt.Errorf("failed to perform HTTP request: %w", err)}
 	}
 	defer res.Body.Close()
```

Comment:
Are there any errors that need to be caught on closing a request body?

Reply:
https://stackoverflow.com/a/47294087/2586553

> There are two things to consider: What would you do with it if you checked it and there was an error? And, what would the side-effects be if there was an error?

> In most cases, for closing a response body, the answer to both questions is... absolutely nothing. If there's nothing you'd do if there was an error and the error has no appreciable impact, there's no reason to check it.